src/native/custom_key_op_handler.c

```diff
@@ -52,23 +82,15 @@ static void s_aws_custom_key_op_handler_perform_operation(
      * function. This function will also catch any exceptions and clear the operation
      * with an exception should it occur.
      */
-    (*env)->CallVoidMethod(
-        env, op_handler->jni_custom_key_op, tls_key_operation_handler_properties.perform_operation_id, jni_operation);
-
-    exception_occured = aws_jni_check_and_clear_exception(env);
-    if (exception_occured) {
-        AWS_LOGF_ERROR(
-            AWS_LS_COMMON_IO,
-            "java_custom_key_op_handler=%p do_operation: Exception occured in Java code",
-            (void *)op_handler);
-        /**
-         * Free the TlsKeyOperation in Java so it doesn't stick around.
-         * Will also complete the key operation
-         */
-        (*env)->CallVoidMethod(env, jni_operation, tls_key_operation_properties.force_close_id);
-    } else {
-        success = true;
-    }
+    (*env)->CallStaticVoidMethod(
+        env,
+        tls_key_operation_properties.cls,
+        tls_key_operation_properties.invoke_operation_id,
+        op_handler->jni_custom_key_op,
+        jni_operation);
+    /* This should never fail, but just to be extra sure... */
+    AWS_FATAL_ASSERT(!aws_jni_check_and_clear_exception(env));
```

Comment:
trivial: 'should never fail because the function we're calling wraps the user callback in a try/catch"